idea/idea-frontend-fir/src/org/jetbrains/kotlin/idea/frontend/api/fir/symbols/KtFirSymbolProvider.kt

```diff
@@ -86,7 +86,9 @@ internal class KtFirSymbolProvider(
     }
 
     override fun getAnonymousFunctionSymbol(psi: KtFunctionLiteral): KtAnonymousFunctionSymbol = withValidityAssertion {
-        psi.withFirDeclarationOfType<FirAnonymousFunction, KtAnonymousFunctionSymbol>(resolveState) {
+        val lambdaExpression = psi.parent as? KtLambdaExpression
```

Comment:
Initially, I tried to revert https://github.com/JetBrains/kotlin/commit/c24ad0ba51888df0aff8a72ee09a386049c760dd, but as its commit message said, it makes sense to use `KtFunctionLiteral`, a subtype of `KtDeclaration`, as part of `getSymbol(KtDeclaration)`. Instead, using `parent` here (with a sufficient check) seems good enough. Let me know what you think.

Reply:
Ah, never mind. I dug further `withFirDeclarationOfType` and found what should have been changed together with https://github.com/JetBrains/kotlin/commit/c24ad0ba51888df0aff8a72ee09a386049c760dd.